### ISSUE TITLE
Camlp5 won't compile (PR#6243 will fix it)

### ISSUE THREAD

**Issue by keigoi** [ocaml/opam-repository#2152]
Camlp5 won't compile with my latest Xcode 5.1.1 (clang 503.0.40).
A dummy line number -1 is generated by camlp5, then it is propagated to the .loc directives in .s and clang assembler won't accept it.

Fortunately, PR#6243 of OCaml http://caml.inria.fr/mantis/file_download.php?file_id=1092&type=bug solves this issue.

Below is the patch for the repo. Please merge this.
```
diff --git a/compilers/4.01.0/4.01.0/4.01.0.comp b/compilers/4.01.0/4.01.0/4.01.0.comp
index 518f13e..8673c46 100644
--- a/compilers/4.01.0/4.01.0/4.01.0.comp
+++ b/compilers/4.01.0/4.01.0/4.01.0.comp
@@ -1,7 +1,10 @@
 opam-version: "1"
 version: "4.01.0"
 src: "http://caml.inria.fr/pub/distrib/ocaml-4.01/ocaml-4.01.0.tar.gz"
-patches:["https://github.com/diml/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"]
+patches:[
+  "https://github.com/diml/ocaml/compare/bd7fa181cb64742c3b6cbb8ee13436554eb18cd7...fix-clang-build.diff"
+  "http://caml.inria.fr/mantis/file_download.php?file_id=1092&type=bug"
+]
 build: [
   ["./configure" "-prefix" prefix "-with-debug-runtime"]
   [make "world"]
-- 
```

Console output:
```
$ opam install camlp5
The following actions will be performed:
 - install camlp5.6.11
1 to install | 0 to reinstall | 0 to upgrade | 0 to downgrade | 0 to remove

=-=-= Installing camlp5.6.11 =-=-=
Building camlp5.6.11:
  ./configure --prefix /Users/keigoi/.opam/4.01.0 -libdir /Users/keigoi/.opam/4.01.0/lib -mandir /Users/keigoi/.opam/4.01.0/man
  make world.opt
  make install
[ERROR] The compilation of camlp5.6.11 failed.
Removing camlp5.6.11.


===== ERROR while installing camlp5.6.11 =====
# opam-version 1.1.1
# os           darwin
# command      make world.opt
# path         /Users/keigoi/.opam/4.01.0/build/camlp5.6.11
# compiler     4.01.0
# exit-code    2
# env-file     /Users/keigoi/.opam/4.01.0/build/camlp5.6.11/camlp5-20646-5f1023.env
# stdout-file  /Users/keigoi/.opam/4.01.0/build/camlp5.6.11/camlp5-20646-5f1023.out
# stderr-file  /Users/keigoi/.opam/4.01.0/build/camlp5.6.11/camlp5-20646-5f1023.err
### stdout ###
# ...[truncated]
# ocamlopt.opt -warn-error A -I ../ocaml_stuff/4.01.0/parsing -c -impl versdep.ppo
# rm -f versdep.ppo
# ocamlrun ../boot/camlp5r -nolib -I ../boot -mode S -o ploc.ppo ploc.ml
# ocamlopt.opt -warn-error A -I ../ocaml_stuff/4.01.0/parsing -c -impl ploc.ppo
# rm -f ploc.ppo
# ocamlrun ../boot/camlp5r -nolib -I ../boot -mode S -o plexing.ppo plexing.ml
# ocamlopt.opt -warn-error A -I ../ocaml_stuff/4.01.0/parsing -c -impl plexing.ppo
# rm -f plexing.ppo
# ocamlrun ../boot/camlp5r -nolib -I ../boot -mode S -o plexer.ppo plexer.ml
# ocamlopt.opt -warn-error A -I ../ocaml_stuff/4.01.0/parsing -c -impl plexer.ppo
### stderr ###
# ...[truncated]
#  .loc 2 -1
#         ^
# /var/folders/_7/gzxd4_kj2kv8c9b5ky6w05lh0000gn/T/camlasm4bb160.s:13099:9: error: unexpected token in '.loc' directive
#  .loc 2 -1
#         ^
# File "plexer.ml", line 1:
# Error: Assembler error, input left in file /var/folders/_7/gzxd4_kj2kv8c9b5ky6w05lh0000gn/T/camlasm4bb160.s
# make[2]: *** [plexer.cmx] Error 2
# make[1]: *** [opt] Error 2
# make: *** [world.opt] Error 2

'opam install camlp5' failed.
```

**Comment by whitequark** [comment]
As I understand, this is fixed in 4.02. Can you confirm?

**Comment by keigoi** [comment]
@whitequark Thanks. Confirmed with Xcode 6.1.1 with OCaml 4.02. Closing.
